crates/bevy_ecs/src/reflect.rs

```diff
@@ -117,20 +118,20 @@ impl ReflectComponent {
         entity: Entity,
     ) -> Option<Mut<'a, dyn Reflect>> {
         // SAFETY: unique world access
-        unsafe { (self.0.reflect_mut)(world, entity) }
+        unsafe { (self.0.reflect_mut)(world.as_interior_mutable(), entity) }
     }
 
     /// # Safety
     /// This method does not prevent you from having two mutable pointers to the same data,
     /// violating Rust's aliasing rules. To avoid this:
-    /// * Only call this method in an exclusive system to avoid sharing across threads (or use a
-    ///   scheduler that enforces safe memory access).
+    /// * Only call this method with a [`InteriorMutableWorld`] that may be used to mutably access the component on the entity `entity`
```

Comment:
Nit: Typo and wording
```suggestion
    /// * Only call this method with an [`InteriorMutableWorld`] that may be used to mutably access the component on `entity`.
```